telemetry/csharp/README.md

```diff
@@ -0,0 +1,22 @@
+# C# AWS Toolkit Telemetry Generator
+
+This generator produces code that generates and records telemetry events. It is intended for use in C# systems like the AWS Toolkit for Visual Studio.
+
+At this time, the generated code does not function in a standalone capacity. It requires datatypes (within a `Amazon.ToolkitTelemetry` namespace) that are produced when an SDK-style Telemetry Client is generated. This code lives within the Toolkit repo at this time. See Roadmap for future plans in this space.
+
+## Generator Usage
+
+See [Options file](ToolkitTelemetryGenerator/Options.cs) for details.
+
+By default, the program produces generated code into the directory where the program is run from.
+
+## Integrating Generated code into the AWS Toolkit for Visual Studio
+
+1. Sync this repo
+1. Build and run ToolkitTelemetryGenerator
+1. Take the generated file (GeneratedCode.cs), and place it in the toolkit, as `toolkitcore/AWSToolkit.Util/Telemetry/Telemetry.cs`
```

Comment:
Anyway we can automate this? As we're adding new telemetry it might be painful to have to manually copy files around.

Reply:
Yes. I felt there was value in a more incremental approach, where we get generation up and working properly first. That is why I created a "roadmap" section in this file. (there are tasks on the backlog along these lines as well)

Reply:
Documenting this behaviour makes me concerned that it'll live longer than it should - moving to automated generation should be part of the definition of 'complete' for this feature. I'm all for breaking things up into pieces - but we should not release this telemetry until it's automated.

Reply:
Discussed offline. The generated code is being integrated into a feature branch, and the work to progress this towards an automated, generated package can be performed incrementally.